debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -0,0 +1,263 @@
+                                              ErrorHandler errorHandler, Clock clock, OracleDatabaseSchema schema,
+                                              OracleTaskContext taskContext) {
+        this.jdbcConnection = jdbcConnection;
+        this.dispatcher = dispatcher;
+        this.clock = clock;
+        this.schema = schema;
+        this.offsetContext = offsetContext;
+        OracleChangeRecordValueConverter converters = new OracleChangeRecordValueConverter(jdbcConnection);
+        this.connectorConfig = connectorConfig;
+        this.catalogName = (connectorConfig.getPdbName() != null) ? connectorConfig.getPdbName() : connectorConfig.getDatabaseName();
+        this.dmlParser = new SimpleDmlParser(catalogName, connectorConfig.getSchemaName(), converters);
+        this.strategy = connectorConfig.getLogMiningStrategy();
+        this.isContinuousMining = connectorConfig.isContinuousMining();
+        this.errorHandler = errorHandler;
+        this.taskContext = taskContext;
+    }
+
+    /**
+     * This is the loop to get changes from LogMiner
+     *
+     * @param context change event source context
+     */
+    @Override
+    public void execute(ChangeEventSourceContext context) {
+        Metronome metronome;
+
+        this.transactionalBufferMetrics = new TransactionalBufferMetrics(taskContext);
+        this.transactionalBufferMetrics.register(LOGGER);
+
+        this.transactionalBuffer = new TransactionalBuffer(connectorConfig.getLogicalName(), errorHandler, transactionalBufferMetrics, connectorConfig.getMaxQueueSize());
+
+        this.logMinerMetrics = new LogMinerMetrics(taskContext);
+        this.logMinerMetrics.register(LOGGER);
+
+        // The top outer loop gives the resiliency on the network disconnections. This is critical for cloud deployment.
+        while (context.isRunning()) {
+            try (Connection connection = jdbcConnection.connection(false);
+                    PreparedStatement fetchFromMiningView = connection
+                            .prepareStatement(SqlUtils.queryLogMinerContents(connectorConfig.getSchemaName(), jdbcConnection.username(), schema))) {
+
+                startScn = offsetContext.getScn();
+                createAuditTable(connection);
+                LOGGER.trace("current millis {}, db time {}", System.currentTimeMillis(), getTimeDifference(connection));
+                transactionalBufferMetrics.setTimeDifference(new AtomicLong(getTimeDifference(connection)));
+
+                if (!isContinuousMining && startScn < getFirstOnlineLogScn(connection)) {
+                    throw new RuntimeException("Online REDO LOG files don't contain the offset SCN. Clean offset and start over");
+                }
+
+                // 1. Configure Log Miner to mine online redo logs
+                setNlsSessionParameters(jdbcConnection);
+                checkSupplementalLogging(jdbcConnection, connectorConfig.getPdbName());
```

Comment:
This could be done once at startup and does not have to be done every time again.

Reply:
I agree.  We have some more changes coming from the contributor so I'll temporarily add a todo for this and once those other contributions are in, we can rework this.

Reply:
sure

Reply:
@Naros, can you log a follow-up?

Reply:
Logged as [DBZ-2531](https://issues.redhat.com/browse/DBZ-2531).